distributed/client.py

```diff
@@ -3448,6 +3453,9 @@ def compute(
             )
 
         variables = [a for a in collections if dask.is_dask_collection(a)]
+        metadata = SpanMetadata(
+            {"collections": [get_collections_metadata(v) for v in variables]}
+        )
```

Comment:
nit: I'd recommend being consistent in how we initialize the span metadata object.

Right now, I see us either passing a plain `dict` to `_graph_to_futures` or calling `SpanMetadata(...)` with a dictionary object.

Personally, I'd prefer calling `SpanMetadata` with the appropriate keyword arguments, i.e, `SpanMetadata(collections=[get_collections_metadata(v) for v in variables])` but I leave that up to you.
```suggestion
        metadata = SpanMetadata(
            collections=[get_collections_metadata(v) for v in variables]
        )
```

Reply:
For additional context: Since `_graph_to_futures` is itself not typed, explicitly calling `SpanMetadata(...)` every time makes it clearer what accepted inputs are for the `span_metadata` keyword argument.